Windows failing to run

Reported that it does not work properly on windows. Pushed 631d596d48ea729b8d7dafd257cfcb1b68e35d1f to avoid using linux specific mappings. Will test and debug to get it working properly on windows this weekend.

Thank you `@Freakman` on discord for testing this!

Current branch is giving this error on windows:
```
zig build-exe llama2 ReleaseSafe native: error: the following command failed with 1 compilation errors:
C:\Users\user\zig\.bin\master\zig.exe build-exe C:\Users\user\Documents\GitHub\llama2.zig\src\main.zig -fstrip -OReleaseSafe --cache-dir C:\Users\user\Documents\GitHub\llama2.zig\zig-cache --global-cache-dir C:\Users\user\AppData\Local\zig --name llama2 --listen=-
Build Summary: 0/3 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
+- install llama2 transitive failure
   +- zig build-exe llama2 ReleaseSafe native 1 errors
C:\Users\user\zig\.bin\master\lib\std\os.zig:102:23: error: root struct of file 'c' has no member named 'MAP'
pub const MAP = system.MAP;
                ~~~~~~^~~~
referenced by:
    main: src\main.zig:619:102
    callMain: C:\Users\user\zig\.bin\master\lib\std\start.zig:614:32
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```
Honestly feels like a zig problem at this point 😅

c:\zig\zig build -Doptimize=ReleaseFast
zig build-exe llama2 ReleaseFast native: error: the following command failed with 1 compilation errors:
c:\zig\zig.exe build-exe C:\!DEV\eh\qlora\llama2.zig\src\main.zig -fstrip -OReleaseFast --cache-dir C:\!DEV\eh\qlora\llama2.zig\zig-cache --global-cache-dir C:\Users\eh\AppData\Local\zig --name llama2 --listen=-
Build Summary: 0/3 steps succeeded; 1 failed (disable with --summary none)
install transitive failure
└─ install llama2 transitive failure
   └─ zig build-exe llama2 ReleaseFast native 1 errors
c:\zig\lib\std\os.zig:99:23: error: root struct of file 'c' has no member named 'MAP'
pub const MAP = system.MAP;
                ~~~~~~^~~~
referenced by:
    main: src\main.zig:844:102
    callMain: c:\zig\lib\std\start.zig:574:32
    remaining reference traces hidden; use '-freference-trace' to see all reference traces

--- a/README.md
+++ b/README.md
@@ -78,7 +78,7 @@ language implementations (no OpenBlas, etc.).
 
 | Implementation                                      | Tokens/s |
 | --------------------------------------------------- | -------- |
-| llama2.zig (this repo)                              | 606      |
+| llama2.zig (this repo)                              | 612      |
 | llama2.c `make runfast -march=native`               | 548      |
 | [llama2.zig](https://github.com/clebert/llama2.zig) | 496      |
 | llama2.c `make run -march=native`                   | 122      |
@@ -92,7 +92,7 @@ language implementations (no OpenBlas, etc.).
 
 | Implementation                        | Tokens/s |
 | ------------------------------------- | -------- |
-| llama2.zig (this repo)                | 504      |
+| llama2.zig (this repo)                | 579      |
 | llama2.c `make runfast -march=native` | 241      |
 
 ## Multi-threaded

--- a/src/main.zig
+++ b/src/main.zig
@@ -842,11 +842,21 @@ pub fn main() !void {
     log("SIMD vector size: {d}\n", .{DEFAULT_VECTOR_WIDTH});
     log("\n", .{});
 
-    // mmap the checkpoint to directly map the weights
-    const mapped_checkpoint = try (std.fs.cwd().openFile(bin_path.?, .{}));
+    const mapped_checkpoint = try std.fs.cwd().openFile(bin_path.?, .{});
     defer mapped_checkpoint.close();
-    const data: []align(mem.page_size) u8 = try std.os.mmap(null, file_size, std.os.PROT.READ, std.os.MAP.PRIVATE, mapped_checkpoint.handle, 0);
-    defer std.os.munmap(data);
+    const data: []align(mem.page_size) u8 = blk: {
+        const buffer = try allocator.alignedAlloc(u8, mem.page_size, file_size);
+        const read_len = try mapped_checkpoint.readAll(buffer);
+        if (read_len != file_size) {
+            std.debug.print("error: failed to read checkpoint file\n", .{});
+            std.process.exit(1);
+        }
+        break :blk buffer;
+        // mmap seems slower
+        // break :blk try std.os.mmap(null, file_size, std.os.PROT.READ, std.os.MAP.PRIVATE, mapped_checkpoint.handle, 0);
+    };
+    defer allocator.free(data);
+
     const weights = Weights.init(&config, data[@sizeOf(ConfigReader)..], shared_weights);
 
     // load the tokens for the model

